### PR TITLE
[REVIEW] New build process script changes

### PR DESCRIPTION
**Pull request by raydouglass**
This PR outlines the changes necessary for the new build process (aka Project Flash) to work with RMM. This PR is necessary to begin shadow testing of the new build process.

The goal is to simultaneously support both the old and new build processes. CI is configured to activate the new process by setting `PROJECT_FLASH=1`. Additionally `BUILD_LIBRMM` and `BUILD_RMM` will be set to `1` as needed by CI in the new process.

CI will stage build artifacts for python and GPU builds in `ci/artifacts/rmm/cpu/` which the build scripts will then consume.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: ci/gpu/build.sh [hunk 1]
```diff
@@ -55,29 +55,71 @@ gcc --version
...
+    done
 
-    cd "${WORKSPACE}/build"
-    GTEST_OUTPUT="xml:${WORKSPACE}/test-results/" make -j${PARALLEL_LEVEL} test
-
-    logger "Python py.test for librmm_cffi..."
     cd $WORKSPACE/python
+    
+    logger "Installing librmm..."
+    conda install -c $WORKSPACE/ci/artifacts/rmm/cpu/conda-bld/ librmm
```
Comment (kkraus14) [inline ci/gpu/build.sh:110]: Would suggest using `--override-channels` or `--strict-channel-priority` here to ensure it doesn't fall back to finding `librmm` in the `rapidsai-nightly` or `rapidsai` channels

File: build.sh [hunk 2]
```diff
@@ -131,11 +131,10 @@ fi
 
 # Build and install the rmm Python package
 if (( NUMARGS == 0 )) || hasArg rmm; then
-    ensureCMakeRan
     cd "${REPODIR}/python"
     if [[ ${INSTALL_TARGET} != "" ]]; then
         echo "building rmm..."
-        python setup.py build_ext --inplace
+        python setup.py build_ext --inplace --library-dir="${LIBRMM_BUILD_DIR}"
```
Comment (kkraus14) [inline build.sh:137]: Does this just add the location to the existing library search paths in the setup.py? Concern is if it only searches here it won't find the other needed shared libraries by the Cython like `libcudart` or `libcuda`.

File: ci/gpu/build.sh (hunk 1 above)
Comment (raydouglass) [reply ci/gpu/build.sh:110]: I don't think either is necessary. Conda install first searches the `-c` channels, then `.condarc` channels. So we're guaranteed to find `librmm` in the local channel first. We need the `.condarc` channels from the container to get an dependencies, so I don't want to use `--override-channels`. And similarly, I don't want `--strict-channel-priority` to mess with the dependency resolution.

File: build.sh (hunk 2 above)
Comment (raydouglass) [reply build.sh:137]: This add to the library search path, IE, an additional `-L` to the compiler/linker.

**Pull request merged**